docs/docs/developers/versions-updating.md

```diff
@@ -0,0 +1,144 @@
+```
+
+```
+57a58
+>         note_type_id: Field,
+```
+:::tip
+
+### Language server version (aztec-nargo)
+The [Noir LSP](https://docs.aztec.network/developers/contracts/main#install-noir-lsp-recommended) uses your local version of `aztec-nargo`, and thus also `aztec-nargo compile`.
+The path of the former (once installed) can be seen by hovering over "Nargo" in the bottom status bar of VS Code, and the latter via the `which aztec-nargo` command.
+
+:::caution
+For Aztec contract files, this should be `aztec-nargo` and for noir-only files this should be `nargo`. Mismatching tools and file types will generate misleading syntax and compiler errors.
+:::caution
+
+This can present confusion when opening older contracts (and dependencies) written in older version of noir, such as:
+- Logs filled with errors from the dependencies
+- Or the LSP fails (re-runs automatically then stops)
+The second point requires a restart of the extension, which you can trigger with the command palette (Ctrl + Shift + P) and typing "Reload Window".
+
+## Updating
+### TL;DR
+
+1. Updating the sandbox and CLI:
+
+```shell
+aztec-up
+```
+
+2. Updating aztec-nr and individual @aztec dependencies:
+
+Inside your project run:
+
+```shell
+cd your/aztec/project
+aztec-cli update . --contract src/contract1 --contract src/contract2
+```
+
+The sandbox must be running for the update command to work. Make sure it is [installed and running](../developers/sandbox/references/sandbox-reference.md).
+
+3. Refer [Migration Notes](../misc/migration_notes.md) on any breaking changes that might affect your dapp
+
+---
+
+There are four components whose versions need to be kept compatible:
+
+1. Aztec Sandbox
+2. Aztec CLI
+3. aztec-nargo
+4. `Aztec.nr`, the Noir framework for writing Aztec contracts
+
+First three are packaged together in docker and are kept compatible by running `aztec-up`.
+But you need to update your Aztec.nr version manually or using `aztec-cli update`.
+
+## Updating Aztec.nr packages
+
+### Automatic update
+
+`aztec-cli` will update your Aztec.nr packages to the appropriate version with the `aztec-cli update` command. Run this command from the root of your project and pass the paths to the folders containing the Nargo.toml files for your projects like so:
+
+```shell
+aztec-cli update . --contract src/contract1 --contract src/contract2
+```
+
+### Manual update
+
+To update the aztec.nr packages manually, update the tags of the `aztec.nr` dependencies in the `Nargo.toml` file.
+
+```diff
+[dependencies]
+-aztec = { git="https://github.com/AztecProtocol/aztec-packages", tag="aztec-packages-v0.7.5", directory="noir-projects/aztec-nr/aztec" }
++aztec = { git="https://github.com/AztecProtocol/aztec-packages", tag="#include_aztec_version", directory="noir-projects/aztec-nr/aztec" }
```

Comment:
I know this was here before, but is this `#include_aztec_version` part of a template script, or is it a placeholder? if the latter, I would make it more obvious that is a placeholder by being extremely verbose, i.e. `<your-aztec-version-here>`

"show, don't tell" is always better

Reply:
it gets replace with the actual latest version when the doc processor runs